internal/core/metadata/controller/http/device_test.go

```diff
@@ -568,15 +555,41 @@ func TestPatchDevice(t *testing.T) {
 		{"Invalid - not found id", []requests.UpdateDeviceRequest{invalidNotFoundId}, http.StatusMultiStatus, http.StatusNotFound},
 		{"Invalid - not found name", []requests.UpdateDeviceRequest{invalidNotFoundName}, http.StatusMultiStatus, http.StatusNotFound},
 		{"Invalid - no id and name", []requests.UpdateDeviceRequest{invalidNoIdAndName}, http.StatusBadRequest, http.StatusBadRequest},
-		{"Invalid - not found service", []requests.UpdateDeviceRequest{notFoundService}, http.StatusMultiStatus, http.StatusNotFound},
 		{"Invalid - not found profile", []requests.UpdateDeviceRequest{notFoundProfile}, http.StatusMultiStatus, http.StatusNotFound},
 		{"Invalid - invalid protocols", []requests.UpdateDeviceRequest{invalidProtocols}, http.StatusMultiStatus, http.StatusInternalServerError},
+		{"Invalid - device service unavailable", []requests.UpdateDeviceRequest{unavailableService}, http.StatusMultiStatus, http.StatusServiceUnavailable},
 	}
 	for _, testCase := range tests {
 		t.Run(testCase.name, func(t *testing.T) {
 			jsonData, err := json.Marshal(testCase.request)
 			require.NoError(t, err)
 
+			var responseEnvelope types.MessageEnvelope
+			mockMessaging := &messagingMocks.MessageClient{}
+			mockMessaging.On("Publish", mock.Anything, mock.Anything).Return(nil)
```

Comment:
Same question on need to mock Publish since code being tested never calls it directly